### PR TITLE
feat(round-settings): fix round time editing bug

### LINKED ISSUE #2113: Manager: unable to edit round end date
**Overview**
I am unable to edit the round end date because of a gas estimation bug. This loom was taken on staging but I verified that the same behavior is happening in production.

**Steps To Reproduce**
1. Go to https://rmgitcoin.on.fleek.co/#/round/0xbd84a5c0affbcd4bf285ab20ebecae20f0a31f36?debug=true
2. Click on Round settings -> Dates -> Edit
3. Try to change the round date to LATER than the previous date

**Observed  behavior**
Loom: https://www.loom.com/share/d7fcdc91338347f38598377d5e1ba650

**Screenshots**
Console error: 
Screenshot 2023-08-03 at 8 12 26 PM

### PR DESCRIPTION
**Pull request by vacekj**
## PR checklist

For every PR, make sure that these statements are true:
  - [x] Includes only changes relevant to the original ticket. Significant refactoring needs to be separated.
  - [x] Doesn't contain type casts and non-null assertions.
  - [x] Doesn't add `@ts-ignore`.
  - [x] Doesn't disable lints.
  - [x] Doesn't use `useState` just for computation - use plain variables instead.
  - [x] Splits components into pure components that don't depend on external state or hooks.
  - [x] Avoid embedding components within other components
  - [x] Doesn't propagate optional values without good reason, doesn't mark property values as optional if that doesn't represent reality.
  - [x] Doesn't duplicate existing code.
  - [x] Parses out-of-domain data - this includes user input, API respones, on-chain data etc.
  - [x] Doesn't contain commented out code.
  - [x] Doesn't contain skipped or empty tests.
  - [x] If this PR adds/updates any feature, it adds/updates its test script

Subjective - at the discretion of the reviewers
- Does things as simply as possible, but not simpler.
- Doesn't reinvent the wheel or create premature abstractions.

##### Refers/Fixes

fixes #2113 #1982

### CODE REVIEW

File: packages/round-manager/src/context/round/UpdateRoundContext.tsx [hunk 1]
```diff
@@ -157,13 +158,22 @@ const _updateRound = async ({
...
     }
 
+    /* Special case - if the round has already started, and we are editing times,
+     * we need to set newApplicationsStartTime and newRoundStartTime to something bigger than the block timestamp.
+     * This won't actually update the values, it's done just to pass the checks in the contract
+     * (and to confuse the developer). */
+    if (Date.now() > round.roundStartTime.getTime()) {
+      round.applicationsStartTime = subSeconds(round.applicationsEndTime, 1);
+      round.roundStartTime = subSeconds(round.applicationsEndTime, 1);
+    }
```
Comment (gravityblast) [inline packages/round-manager/src/context/round/UpdateRoundContext.tsx:168]: let's explain why in the comment, it makes sense to remember that the contract call doesn't fail, 
it updates the end time only if the round hasn't started yet

**Review by gravityblast — APPROVED** [review]

File: packages/round-manager/src/context/round/UpdateRoundContext.tsx (hunk 1 above)
Comment (vacekj) [reply packages/round-manager/src/context/round/UpdateRoundContext.tsx:168]: Could you be more specific here? Not sure what you want to change about the explanation.
Feel free to suggest the changes to be commited directly in the review here.

Comment (vacekj) [reply packages/round-manager/src/context/round/UpdateRoundContext.tsx:168]: TODO: link the contract checks here

File: packages/round-manager/src/context/round/UpdateRoundContext.tsx [hunk 2]
```diff
@@ -157,13 +158,22 @@ const _updateRound = async ({
       transactionBuilder.add(UpdateAction.UPDATE_ROUND_FEE_PERCENTAGE, [arg]);
     }
 
+    /* Special case - if the round has already started, and we are editing times,
+     * we need to set newApplicationsStartTime and newRoundStartTime to something bigger than the block timestamp.
+     * This won't actually update the values, it's done just to pass the checks in the contract
+     * (and to confuse the developer). */
```
Comment (vacekj) [inline packages/round-manager/src/context/round/UpdateRoundContext.tsx:164]:
```suggestion
     * (and to confuse the developer). 
    *  https://github.com/allo-protocol/allo-contracts/blob/9c50f53cbdc2844fbf3cfa760df438f6fe3f0368/contracts/round/RoundImplementation.sol#L339C1-L339C1*/
```

**Review by bhargavaparoksham — APPROVED** [review]

**Pull request merged**